examples/java/src/test/java/org/apache/beam/examples/WordCountIT.java

```diff
@@ -46,9 +68,121 @@
-        String.format("WordCountIT-%tF-%<tH-%<tM-%<tS-%<tL", new Date()), "output", "results"}));
+    options.setOutput(
+        Joiner.on("/")
+            .join(
+                new String[] {
+                  options.getTempRoot(),
+                  String.format("WordCountIT-%tF-%<tH-%<tM-%<tS-%<tL", new Date()),
+                  "output",
+                  "results"
+                }));
+    options.setOnSuccessMatcher(new WordCountOnSuccessMatcher(options.getOutput() + "*"));
 
     WordCount.main(TestPipeline.convertToArgs(options));
   }
+
+  /**
+   * Matcher for verifying WordCount output data.
+   */
+  static class WordCountOnSuccessMatcher extends BaseMatcher<PipelineResult>
+      implements SerializableMatcher<PipelineResult> {
+
+    private static final Logger LOG = LoggerFactory.getLogger(WordCountOnSuccessMatcher.class);
+
+    private static final String EXPECTED_CHECKSUM = "C780E9466B8635AF1D11B74BBD35233A82908A02";
+
+    private final String outputPath;
+
+    WordCountOnSuccessMatcher(String outputPath) {
+      this.outputPath = outputPath;
+    }
+
+    @Override
+    public boolean matches(Object o) {
+      if (o == null || !(o instanceof PipelineResult)) {
+        fail(String.format("Expected PipelineResult but received %s", o));
+      }
+
+      if (outputPath == null || outputPath.isEmpty()) {
+        fail(String.format("Expected valid output path, but received %s", outputPath));
+      }
+
+      //Load output data
+      LOG.info("Loading actual from path: {}", outputPath);
+      List<String> outputs = read(outputPath);
+
+      // Verify checksum of outputs
+      Collections.sort(outputs);
+      try {
+        String checksum = generateChecksum(outputs);
+        LOG.info("Generate checksum for output data: {}", checksum);
+
+        return checksum.equals(EXPECTED_CHECKSUM);
+      } catch (NoSuchAlgorithmException e) {
+        e.printStackTrace();
+      }
+
+      return false;
+    }
+
+    private List<String> read(String path) {
+      List<String> readData = new ArrayList<>();
+
+      try {
+        IOChannelFactory factory = IOChannelUtils.getFactory(path);
+
+        // Match inputPath which may contains glob
+        Collection<String> files = factory.match(path);
+
+        // Read data from file paths
+        for (String file : files) {
+          Reader reader = Channels.newReader(factory.open(file), StandardCharsets.UTF_8.name());
+          LineReader lineReader = new LineReader(reader);
+
+          String line;
+          while ((line = lineReader.readLine()) != null) {
+            readData.add(line);
+          }
+
+          reader.close();
+        }
+      } catch (IOException e) {
+        e.printStackTrace();
+      }
+
+      return readData;
+    }
+
+    /**
+     * Generate checksum of a string collection using SHA-1 algorithm.
+     */
+    private String generateChecksum(List<String> lines) throws NoSuchAlgorithmException {
```

Comment:
If you use the guava hashfunction, you won't have to worry about the NoSuchAlgorithmException since sha1 is declared there.